frame/contracts/src/rent.rs

```diff
@@ -399,8 +399,11 @@ where
 			&alive_contract_info,
 			code_size,
 		);
+
+		// We skip the eviction in case one is in order.
+		// Evictions should only be perfomed by [`try_eviction`].
 		let new_contract_info = Self::enact_verdict(
-			account, alive_contract_info, current_block_number, verdict, Some(module),
+			account, alive_contract_info, current_block_number, verdict, None,
```

Comment:
So this was a bug you just noticed here?

Reply:
More of an optimization. `compute_projection` is called from RPC where changes to storage are not persisted. This avoids superfluous work.